src/tools/simeng/main.cc

```diff
@@ -197,107 +162,46 @@ int main(int argc, char** argv) {
     //                                  17, 4, 3, 22, 117, 11, 4,  12, 10, 18};
     // memcpy(memory, memoryValues.data(), memoryValues.size() * sizeof(int));
 
-    process = std::make_unique<simeng::kernel::LinuxProcess>(
-        simeng::span<char>(reinterpret_cast<char*>(hex), sizeof(hex)), config);
+    coreInstance->createProcess(
+        simeng::span<char>(reinterpret_cast<char*>(hex), sizeof(hex)));
   }
 
-  // Read the process image and copy to memory
-  auto processImage = process->getProcessImage();
-  size_t processMemorySize = processImage.size();
-  char* processMemory = new char[processMemorySize]();
-  std::copy(processImage.begin(), processImage.end(), processMemory);
-
-  uint64_t entryPoint = process->getEntryPoint();
-
-  // Create the OS kernel with the process
-  simeng::kernel::Linux kernel;
-  kernel.createProcess(*process.get());
-
-  simeng::FlatMemoryInterface instructionMemory(processMemory,
-                                                processMemorySize);
-
-  // Create the architecture, with knowledge of the kernel
-  std::unique_ptr<simeng::arch::Architecture> arch =
-      std::make_unique<simeng::arch::aarch64::Architecture>(kernel, config);
-
-  auto predictor = simeng::GenericPredictor(config);
-  auto config_ports = config["Ports"];
-  std::vector<std::vector<uint16_t>> portArrangement(config_ports.size());
-  // Extract number of ports
-  for (size_t i = 0; i < config_ports.size(); i++) {
-    auto config_groups = config_ports[i]["Instruction-Group-Support"];
-    // Extract number of groups in port
-    for (size_t j = 0; j < config_groups.size(); j++) {
-      portArrangement[i].push_back(config_groups[j].as<uint16_t>());
-    }
+  // Get simualtion mode and data memory interface type
+  simeng::SimulationMode mode = coreInstance->getSimulationMode();
+  simeng::MemInterfaceType L1Dtype = simeng::MemInterfaceType::Flat;
+  std::string modeString = "Emulation";
+  if (mode == simeng::SimulationMode::OutOfOrder) {
+    modeString = "Out-of-Order";
+    L1Dtype = simeng::MemInterfaceType::Fixed;
   }
-  auto portAllocator = simeng::pipeline::BalancedPortAllocator(portArrangement);
-
-  // Configure reservation station arrangment
-  std::vector<std::pair<uint8_t, uint64_t>> rsArrangement;
-  for (size_t i = 0; i < config["Reservation-Stations"].size(); i++) {
-    auto reservation_station = config["Reservation-Stations"][i];
-    for (size_t j = 0; j < reservation_station["Ports"].size(); j++) {
-      uint8_t port = reservation_station["Ports"][j].as<uint8_t>();
-      if (rsArrangement.size() < port + 1) {
-        rsArrangement.resize(port + 1);
-      }
-      rsArrangement[port] = {i, reservation_station["Size"].as<uint16_t>()};
-    }
+  if (mode == simeng::SimulationMode::InOrderPipelined) {
+    modeString = "In-Order Pipelined";
   }
 
-  int iterations = 0;
+  // Create memory interfaces
+  std::shared_ptr<simeng::MemoryInterface> dataMemory =
```

Comment:
Also since we already do have a `setL1DataMemory function` which allows us to set any memory interface shouldn't`createMemoryL1DataMemory` create a memory interface associated with a mode? e.g OoO -> Fixed

Reply:
I believe https://github.com/UoB-HPC/SimEng/pull/247#discussion_r970681853 answers this query.